contracts/Collection.sol

```diff
@@ -0,0 +1,119 @@
+    /** Royalty manager role.
+     *  Used for configuring the amounts and beneficiaries of royalties on primary and secondary transfers of this NFT.
+     */
+    bytes32 public constant ROYALTY_MANAGER_ROLE = keccak256("ROYALTY_MANAGER_ROLE");
+    // The address of Freeport contract.
+    Freeport public freeport;
+    /** A counter of NFT types issued.
+     * This is used to generate unique NFT IDs.
+     */
+    uint32 public idCounter;
+
+    function setName(string memory newName) public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "only admin");
+        name = newName;
+    }
+
+    /** Sets Freeport contract address.
+     */
+    function setFreeport(address _freeportContract) public {
+        require(freeport == Freeport(address(0)));
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()));
+
+        freeport = Freeport(_freeportContract);
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId)
+    public view virtual override(BaseNFT) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /// Issuer interface
+    /** Issue NFT of a new type, and return its ID.
+     *
+     * No more NFT of this type can be issued again.
+     *
+     * The caller will be recorded as the issuer and it will initially own the NFT.
+     */
+    function issueNft(bytes memory data)
+    public returns (uint256) {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "only admin");
+        return _issueAs(_msgSender(), data);
+    }
+
+    /** Internal implementation of the function issue.
+     */
+    function _issueAs(address issuer, bytes memory data)
+    internal returns (uint32) {
+        idCounter = idCounter + 1;
+
+        _mint(issuer, idCounter, 1, data);
+
+        return idCounter;
+    }
+    /// Issuer interface
+
+    /// Royalty management interface
+    /** Configure the amounts and beneficiaries of royalties on primary and secondary transfers of this NFT.
+     *  Delegating to Freeport implementation.
+     */
+    function setupRoyaltyConfiguration(
+        address primaryRoyaltyAccount,
+        uint256 primaryRoyaltyCut,
+        uint256 primaryRoyaltyMinimum,
+        address secondaryRoyaltyAccount,
+        uint256 secondaryRoyaltyCut,
+        uint256 secondaryRoyaltyMinimum)
+    public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender())
+            || hasRole(ROYALTY_MANAGER_ROLE, _msgSender()), "only admin or royalty manager");
+
+        for (uint32 i = 1; i <= idCounter; i++) {
+            freeport.configureRoyalties(
+                getGlobalNftId(i),
+                primaryRoyaltyAccount,
+                primaryRoyaltyCut,
+                primaryRoyaltyMinimum,
+                secondaryRoyaltyAccount,
+                secondaryRoyaltyCut,
+                secondaryRoyaltyMinimum
+            );
+        }
+    }
+    /// Royalty management interface
```

Comment:
There should also be a getter for royalties. That will be used by the marketplace.